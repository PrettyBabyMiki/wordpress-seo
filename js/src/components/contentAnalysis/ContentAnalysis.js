/* global wpseoPostScraperL10n */

import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import Results from "./Results";

class ContentAnalysis extends React.Component {
	render() {
		debugger;
		return <Results
			showLanguageNotice={ true }
			changeLanguageLink={ wpseoPostScraperL10n.settings_link }
			language={ wpseoPostScraperL10n.language }
			results={ this.props.results }/>
	}
}

ContentAnalysis.propTypes = {
	results: PropTypes.array,
};

/**
 * Map redux state to ContentAnalysis props.
 *
 * @param {Object} state The redux state.
 *
 * @returns {Object} Props that should be passed to ContentAnalysis.
 */
function mapStateToProps( state ) {
	return {
		results: state.analysis.readability,
	};
}

export default connect( mapStateToProps )( ContentAnalysis );
