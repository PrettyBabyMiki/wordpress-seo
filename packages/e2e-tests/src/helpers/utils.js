/**
 * WordPress e2e utilities
 */
import {
    installPlugin,
    activatePlugin,
    visitAdminPage,
    pressKeyWithModifier,
} from "@wordpress/e2e-test-utils";

import { addQueryArgs } from '@wordpress/url';

export const createCustomPostType = async( slug ) => {
   
}