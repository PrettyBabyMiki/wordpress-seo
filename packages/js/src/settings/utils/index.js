import { filter, isArray, isObject, join, reduce } from "lodash";

/**
 * Determines if the given value should flatten.
 * @param {*} value The value.
 * @returns {boolean} True if the value is an object or an array.
 */
const shouldFlatten = value => isObject( value ) || isArray( value );

/**
 * Flattens nested object to a single level. Props will be keyed by dot syntax path.
 * @param {Object} object The object to flatten.
 * @param {string} parentPath The parent object path.
 * @param {function} getShouldFlatten Determines whether the entry should be flattened.
 * @returns {Object} Flattened object.
 */
export const flattenObject = ( object, parentPath = "", getShouldFlatten = shouldFlatten ) => reduce(
	object,
	( acc, value, key ) => {
		const flatKey = join( filter( [ parentPath, key ], Boolean ), "." );

		return getShouldFlatten( value, key ) ? {
			...acc,
			...flattenObject( value, flatKey, getShouldFlatten ),
		} : {
			...acc,
			[ flatKey ]: value,
		};
	},
	{}
);
