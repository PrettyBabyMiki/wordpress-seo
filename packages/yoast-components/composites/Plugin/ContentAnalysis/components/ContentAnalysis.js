import { ContentAnalysis } from "@yoast/analysis-report";

export default ContentAnalysis;
