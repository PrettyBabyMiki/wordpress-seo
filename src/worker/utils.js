/**
 * Encodes the payload for transfer.
 *
 * @param {Object} payload The payload.
 *
 * @returns {string} Transferable string.
 */
export function encodePayload( payload ) {
	return JSON.stringify( payload );
}

/**
 * Decodes the payload for use.
 *
 * @param {string} payload The encoded payload.
 *
 * @returns {any} The parsed result.
 */
export function decodePayload( payload ) {
	try {
		return JSON.parse( payload );
	} catch( error ) {
		return payload;
	}
}
