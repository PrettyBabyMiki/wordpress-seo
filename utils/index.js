import { makeOutboundLink } from "./makeOutboundLink";
import { getFeed } from "./getFeed";

export default {
	getFeed,
	makeOutboundLink,
};
