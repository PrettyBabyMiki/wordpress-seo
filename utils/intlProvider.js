import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";

export const createComponentWithIntl = ( children, props = { locale: "en" }, renderOptions = {} ) => {
	return renderer.create(
		<IntlProvider { ...props }>
			{ children }
		</IntlProvider>,
		renderOptions
	);
};
