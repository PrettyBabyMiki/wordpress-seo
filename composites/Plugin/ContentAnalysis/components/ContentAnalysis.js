import React from "react";
import styled from "styled-components";
import colors from "../../../../style-guide/colors.json";
import PropTypes from "prop-types";

import { injectIntl, intlShape, defineMessages, FormattedMessage } from "react-intl";
import { makeOutboundLink } from "../../../../utils/makeOutboundLink";
import AnalysisResult from "../components/AnalysisResult.js";
import AnalysisCollapsible from "../components/AnalysisCollapsible.js";

export const ContentAnalysisContainer = styled.div`
	width: 100%;
	background-color: white;
	max-width: 800px;
	margin: 0 auto;
`;

const LanguageNotice = styled.p`
	min-height: 24px;
	padding-top: 8px;
	margin-left: 36px;
`;

const ChangeLanguageLink = makeOutboundLink( styled.a`
	color: ${ colors.$color_blue };
	margin-left: 4px;
` );

const messages = defineMessages( {
	languageNoticeLink: {
		id: "content-analysis.language-notice-link",
		defaultMessage:	"Change language",
	},
	errorsHeader: {
		id: "content-analysis.errors",
		defaultMessage: "Errors",
	},
	problemsHeader: {
		id: "content-analysis.problems",
		defaultMessage: "Problems",
	},
	improvementsHeader: {
		id: "content-analysis.improvements",
		defaultMessage: "Improvements",
	},
	considerationsHeader: {
		id: "content-analysis.considerations",
		defaultMessage: "Considerations",
	},
	goodHeader: {
		id: "content-analysis.good",
		defaultMessage: "Good",
	},
	highlight: {
		id: "content-analysis.highlight",
		defaultMessage: "Highlight this result in the text",
	},
} );

/**
 * Returns the ContentAnalysis component.
 *
 * @returns {ReactElement} The ContentAnalysis component.
 */
class ContentAnalysis extends React.Component {
	/**
	 * The constructor.
	 *
	 * @param {object} props The component props.
	 *
	 * @returns {void}
	 */
	constructor( props ) {
		super( props );

		this.state = {
			checked: "",
		};
	}

	/**
	 * Handles button clicks. Makes sure no more than one button can be active at the same time.
	 *
	 * @param {string} id The button id.
	 * @param {function} marker Function to apply marker to the editor.
	 *
	 * @returns {void}
	 */
	handleClick( id, marker ) {
		let checkedId = id;

		// Uncheck if button is deactivated.
		if ( id === this.state.checked ) {
			checkedId = "";
		}

		// Set state and call onMarkButtonClick callback.
		this.setState( {
			checked: checkedId,
		}, () => {
			this.props.onMarkButtonClick( this.state.checked, marker );
		} );
	}

	/**
	 * Gets the color for the bullet, based on the rating.
	 *
	 * @param {string} rating The rating of the result.
	 *
	 * @returns {string} The color for the bullet.
	 */
	getColor( rating ) {
		switch ( rating ) {
			case "good":
				return colors.$color_good;
			case "OK":
				return colors.$color_ok;
			case "bad":
				return colors.$color_bad;
			default:
				return colors.$color_score_icon;
		}
	}

	/**
	 * Returns an AnalysisResult component for each result.
	 *
	 * @param {array} results The analysis results.
	 *
	 * @returns {array} A list of AnalysisResult components.
	 */
	getResults( results ) {
		return results.map( ( result ) => {
			let color = this.getColor( result.rating );
			return <AnalysisResult
				key={ result.id }
				text={ result.text }
				bulletColor={ color }
				hasMarksButton={ result.hasMarks }
				ariaLabel={ this.props.intl.formatMessage( messages.highlight ) }
				pressed={ result.id === this.state.checked }
				buttonId={ result.id }
				onButtonClick={ this.handleClick.bind( this, result.id, result.marker ) }
			/>;
		} );
	}

	/**
	 * Render the language notice. Provides a link to a setting page in case of administrator, a notice to contact an
	 * administrator otherwise.
	 *
	 * @returns {ReactElement} The rendered language notice.
	 */
	renderLanguageNotice() {
		let showLanguageNotice = this.props.showLanguageNotice;
		let canChangeLanguage = this.props.canChangeLanguage;
		if( ! showLanguageNotice ) {
			return null;
		}
		if( canChangeLanguage ) {
			return (
				<LanguageNotice>
					<FormattedMessage
						id="content-analysis.language-notice"
						defaultMessage="Your site language is set to {language}."
						values={ { language: <strong>{ this.props.language }</strong> } } />
					<ChangeLanguageLink href={ this.props.changeLanguageLink }>
						{ this.props.intl.formatMessage( messages.languageNoticeLink ) }
					</ChangeLanguageLink>
				</LanguageNotice>
			);
		}
		return (
			<LanguageNotice>
				<FormattedMessage
					id="content-analysis.language-notice"
					defaultMessage="Your site language is set to {language}."
					values={ { language: <strong>{ this.props.language }</strong> } } />
				{ " " }
				<FormattedMessage
					id="content-analysis.language-notice-contact-admin"
					defaultMessage="Contact your site administrator, if this is not correct."
					values={ { language: <strong>{ this.props.language }</strong> } } />
			</LanguageNotice>
		);
	}

	/**
	 * Renders a ContentAnalysis component.
	 *
	 * @returns {ReactElement} The rendered ContentAnalysis component.
	 */
	render() {
		let problemsResults = this.props.problemsResults;
		let improvementsResults = this.props.improvementsResults;
		let goodResults = this.props.goodResults;
		let considerationsResults = this.props.considerationsResults;
		let errorsResults = this.props.errorsResults;


		// Analysis collapsibles are only rendered when there is at least one analysis result for that category present.
		return (
			<ContentAnalysisContainer>
				{ this.renderLanguageNotice() }
				{ errorsResults.length > 0 &&
				<AnalysisCollapsible
					hasHeading={ true }
					headingLevel={ 2 }
					initialIsOpen={ true }
					title={ this.props.intl.formatMessage( messages.errorsHeader ) }
				>
					{ this.getResults( errorsResults ) }
				</AnalysisCollapsible> }
				{ problemsResults.length > 0 &&
					<AnalysisCollapsible
						hasHeading={ true }
						headingLevel={ 2 }
						initialIsOpen={ true }
						title={ this.props.intl.formatMessage( messages.problemsHeader ) }
					>
						{ this.getResults( problemsResults ) }
					</AnalysisCollapsible> }
				{ improvementsResults.length > 0 &&
					<AnalysisCollapsible
						hasHeading={ true }
						headingLevel={ 2 }
						title={ this.props.intl.formatMessage( messages.improvementsHeader ) }
					>
						{ this.getResults( improvementsResults ) }
					</AnalysisCollapsible> }
				{ considerationsResults.length > 0 &&
					<AnalysisCollapsible
						hasHeading={ true }
						headingLevel={ 2 }
						title={ this.props.intl.formatMessage( messages.considerationsHeader ) }
					>
						{ this.getResults( considerationsResults ) }
					</AnalysisCollapsible> }
				{ goodResults.length > 0 &&
					<AnalysisCollapsible
						hasHeading={ true }
						headingLevel={ 2 }
						title={this.props.intl.formatMessage( messages.goodHeader ) }
					>
						{ this.getResults( goodResults ) }
					</AnalysisCollapsible> }
			</ContentAnalysisContainer>
		);
	}
}

ContentAnalysis.propTypes = {
	onMarkButtonClick: PropTypes.func,
	problemsResults: PropTypes.array,
	improvementsResults: PropTypes.array,
	goodResults: PropTypes.array,
	considerationsResults: PropTypes.array,
	errorsResults: PropTypes.array,
	changeLanguageLink: PropTypes.string.isRequired,
	canChangeLanguage: PropTypes.bool,
	language: PropTypes.string.isRequired,
	showLanguageNotice: PropTypes.bool,
	intl: intlShape.isRequired,
};

ContentAnalysis.defaultProps = {
	onMarkButtonClick: () => {},
	problemsResults: [],
	improvementsResults: [],
	goodResults: [],
	considerationsResults: [],
	errorsResults: [],
	showLanguageNotice: false,
	canChangeLanguage: false,
};

export default injectIntl( ContentAnalysis );
