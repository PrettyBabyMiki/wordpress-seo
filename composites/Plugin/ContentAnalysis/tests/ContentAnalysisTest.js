import React from "react";
import { createComponentWithIntl } from "../../../../utils/intlProvider";
import ContentAnalysis from "../components/ContentAnalysis.js";

const problemsResults = [
	{
		text: "Your text is bad, and you should feel bad.",
		id: "1",
		rating: "bad",
		hasMarks: true,
	},
];

const goodResults = [
	{
		text: "You're doing great!",
		id: "2",
		rating: "good",
		hasMarks: false,
	},
	{
		text: "Woohoo!",
		id: "3",
		rating: "good",
		hasMarks: true,
	},
];

const improvementsResults = [
	{
		text: "I know you can do better! You can do it!",
		id: "4",
		rating: "OK",
		hasMarks: false,
	},
];

const errorsResults = [
	{
		text: "Error: Analysis not loaded",
		id: "5",
		rating: "feedback",
		hasMarks: false,
	},
];

const considerationsResults = [
	{
		text: "Maybe you should change this...",
		id: "6",
		rating: "feedback",
		hasMarks: false,
	},
];

describe( "ContentAnalysis", () => {
	it( "the ContentAnalysis component without language notice matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ problemsResults }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component without problems matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ [] }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component without problems and improvements matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ [] }
				improvementsResults={ [] }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component without problems, improvements and considerations matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ [] }
				improvementsResults={ [] }
				goodResults={ goodResults }
				considerationsResults={ [] }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component without problems and considerations, but with improvements and good matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ [] }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ [] }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component with specified header level matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				headingLevel={ 3 }
				problemsResults={ problemsResults }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component with language notice matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ problemsResults }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
				showLanguageNotice={ true }
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component with language notice for someone who can change the language matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ problemsResults }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
				showLanguageNotice={ true }
				canChangeLanguage={ true }
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component with language notice for someone who cannot change the language matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ problemsResults }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
				showLanguageNotice={ true }
				canChangeLanguage={ false }
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component with disabled buttons matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ problemsResults }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
				showLanguageNotice={ true }
				marksButtonStatus={ "disabled" }
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component with hidden buttons matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ problemsResults }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
				showLanguageNotice={ true }
				marksButtonStatus={ "hidden" }
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "the ContentAnalysis component with HelpText matches the snapshot", () => {
		const component = createComponentWithIntl(
			<ContentAnalysis
				problemsResults={ problemsResults }
				improvementsResults={ improvementsResults }
				goodResults={ goodResults }
				considerationsResults={ considerationsResults }
				errorsResults={ errorsResults }
				changeLanguageLink={ "#" }
				language="English"
				helpText={ [
					"This is a text to help you ",
					<a key="1" href="http://www.example.com">with a link</a>,
					" and some more text.",
				] }
			/>
		);

		let tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
