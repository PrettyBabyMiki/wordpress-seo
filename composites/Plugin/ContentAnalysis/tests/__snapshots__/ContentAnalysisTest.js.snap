// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ContentAnalysis component matches the snapshot 1`] = `
<div
  className="sc-fjdhpX ekEfiG"
>
  <div
    className="sc-jzJRlG ibMVYq"
  >
    <span>
      Your site language is set to 
      <strong>
        English
      </strong>
      .
    </span>
    <a
      className="sc-cSHVUG fzxnjC"
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      Change language
      <span
        className="sc-bdVaJa hrXKaN"
      >
        (Opens in a new browser tab)
      </span>
    </a>
  </div>
  <div
    className="sc-gZMcBi liHqSy"
  >
    <button
      aria-expanded={true}
      className="sc-gqjmRU nPXTt sc-iwsKbI ljgvpn sc-dnqmqq fzLqCC sc-htoDjs dFKnSN sc-gzVnrw jzmimO sc-bZQynM iYyoWB sc-EHOje cUKXHE"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="sc-kAzzGY ibpeNC sc-chPdSV coNTRN"
        focusable="false"
        role="img"
      />
      <span
        className="sc-VigVT bGMrIs"
      >
        Errors (1)
      </span>
    </button>
    <ul
      className="sc-jTzLTM cTbfwU"
      role="list"
    >
      <li
        className="sc-htpNat gcbOOr"
      >
        <svg
          aria-hidden="true"
          className="sc-bxivhb icogks sc-kgoBCf hlIUOu"
          focusable="false"
          role="img"
        />
        <p
          className="sc-ifAKCX cHWEpb"
        >
          Error: Analysis not loaded
        </p>
      </li>
    </ul>
  </div>
  <div
    className="sc-gZMcBi liHqSy"
  >
    <button
      aria-expanded={true}
      className="sc-gqjmRU nPXTt sc-iwsKbI ljgvpn sc-dnqmqq fzLqCC sc-htoDjs dFKnSN sc-gzVnrw jzmimO sc-bZQynM iYyoWB sc-EHOje cUKXHE"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="sc-kGXeez jtnyJj sc-kpOJdX hmYSjx"
        focusable="false"
        role="img"
      />
      <span
        className="sc-VigVT bGMrIs"
      >
        Problems (1)
      </span>
    </button>
    <ul
      className="sc-jTzLTM cTbfwU"
      role="list"
    >
      <li
        className="sc-htpNat gcbOOr"
      >
        <svg
          aria-hidden="true"
          className="sc-bxivhb icogks sc-dxgOiQ ctIQJE"
          focusable="false"
          role="img"
        />
        <p
          className="sc-ifAKCX cHWEpb"
        >
          Your text is bad, and you should feel bad.
        </p>
        <button
          aria-label="highlight this result in the text"
          aria-pressed={false}
          className="sc-bwzfXH iSOXYD"
          id="1"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="sc-ckVGcZ fSBahx"
            focusable="false"
            role="img"
          />
        </button>
      </li>
    </ul>
  </div>
  <div
    className="sc-gZMcBi liHqSy"
  >
    <button
      aria-expanded={false}
      className="sc-gqjmRU nPXTt sc-iwsKbI ljgvpn sc-dnqmqq fzLqCC sc-htoDjs dFKnSN sc-gzVnrw jzmimO sc-bZQynM iYyoWB sc-EHOje cUKXHE"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="sc-jKJlTe fTuuZL sc-eNQAEJ bmrIlx"
        focusable="false"
        role="img"
      />
      <span
        className="sc-VigVT bGMrIs"
      >
        Improvements (1)
      </span>
    </button>
  </div>
  <div
    className="sc-gZMcBi liHqSy"
  >
    <button
      aria-expanded={false}
      className="sc-gqjmRU nPXTt sc-iwsKbI ljgvpn sc-dnqmqq fzLqCC sc-htoDjs dFKnSN sc-gzVnrw jzmimO sc-bZQynM iYyoWB sc-EHOje cUKXHE"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="sc-hMqMXs gteQSO sc-kEYyzF kVGrFb"
        focusable="false"
        role="img"
      />
      <span
        className="sc-VigVT bGMrIs"
      >
        Considerations (1)
      </span>
    </button>
  </div>
  <div
    className="sc-gZMcBi liHqSy"
  >
    <button
      aria-expanded={false}
      className="sc-gqjmRU nPXTt sc-iwsKbI ljgvpn sc-dnqmqq fzLqCC sc-htoDjs dFKnSN sc-gzVnrw jzmimO sc-bZQynM iYyoWB sc-EHOje cUKXHE"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="sc-kkGfuU bhfWur sc-iAyFgw eGHSYb"
        focusable="false"
        role="img"
      />
      <span
        className="sc-VigVT bGMrIs"
      >
        Good (2)
      </span>
    </button>
  </div>
</div>
`;
